warehouse/ingest-core/src/main/java/datawave/ingest/data/normalizer/SimpleGroupFieldNameParser.java

```diff
@@ -35,20 +30,37 @@ public SimpleGroupFieldNameParser(boolean shouldIncludeDelimInGroupName) {
      */
     public NormalizedContentInterface extractFieldNameComponents(NormalizedContentInterface origField) {
         String originalIndexedFieldName = origField.getIndexedFieldName();
-        String normParts[] = StringUtils.split(originalIndexedFieldName, '.');
+        String[] splits = StringUtils.split(originalIndexedFieldName, '.');
         int index = originalIndexedFieldName.indexOf('.');
         
-        if (index > -1) {
-            String baseFieldName = originalIndexedFieldName.substring(0, index);
-            String group = originalIndexedFieldName.substring(index + (shouldIncludeDelimInGroupName ? 0 : 1));
+        if (index > 1) {
+            // we are nested
             
             NormalizedFieldAndValue revisedField = new NormalizedFieldAndValue(origField);
             
-            // we are nested
+            String baseFieldName = null;
+            String group = null;
+            String subgroup = null;
+            
+            baseFieldName = splits[0];
+            group = originalIndexedFieldName.substring(index + 1);
+            
+            if (parentGroupingMatch.matcher(group).matches()) {
```

Comment:
since we went through the trouble of splitting the string on dots, maybe simply pulling off every thing after the _ on each token is sufficient and then simply merge those tokens back together again for the "group".

Reply:
the problem is that sometimes we have PARENT_FIELD_NAME, sometimes PARENT_FIELD_NAME_0, and PARENT_0, so we can't be that lax with everything after the last _

Reply:
I figured I was missing something.  I really don't get this comment.  Can you give me an example?

Reply:
oh, I see what you mean.  ok then check for digits as I did in the trim routine.   but we still can use the tokens to rebuild.  Just a suggestion.